docs/hazmat/primitives/cryptographic-hashes.rst

```diff
@@ -62,11 +62,67 @@ Message digests (Hashing)
         Finalize the current context and return the message digest as bytes.
 
         After ``finalize`` has been called this object can no longer be used
-        and :meth:`update`, :meth:`copy`, and :meth:`finalize` will raise an
+        and :meth:`.update`, :meth:`.copy`, and :meth:`.finalize` will raise an
         :class:`~cryptography.exceptions.AlreadyFinalized` exception.
 
         :return bytes: The message digest as bytes.
 
+.. class:: XOFHash(algorithm)
+
+    An extendable output function (XOF) is a cryptographic hash function that
+    can produce an arbitrary amount of output for a given input. The output
+    can be obtained by repeatedly calling :meth:`.squeeze` with the desired
+    length.
+
+    .. doctest::
+
+        >>> import sys
+        >>> from cryptography.hazmat.primitives import hashes
+        >>> digest = hashes.XOFHash(hashes.SHAKE128(digest_size=sys.maxsize))
+        >>> digest.update(b"abc")
+        >>> digest.update(b"123")
+        >>> digest.squeeze(32)
+        b'\x18\xd6\xbd\xeb5u\x83[@\xfa%/\xdc\xca\x9f\x1b\xc2\xeb\x12\x05\xc3\xf9Bu\x88\xe0\xda\x80FvAV'
+
+    :param algorithm: A
+        :class:`~cryptography.hazmat.primitives.hashes.ExtendableOutputFunction`
+        instance such as those described
+        :ref:`below <extendable-output-functions>`. The ``digest_size``
+        parameter for :class:`SHAKE128` and :class:`SHAKE256`
+        is the maximum number of bytes that can be squeezed from the XOF when
+        using this class.
```

Comment:
I'd describe it as "the digest size passed to the has is the maximum..." don't make it specific to SHAKE